lib/Collection.js

```diff
@@ -0,0 +1,193 @@
+    }
+    const {controllerKey, instance, capability} = this;
+    const edvDoc = await getEdvDocument({controllerKey, instance, capability});
+    const id = await EdvClient.generateId();
+    await edvDoc.write({doc: {id, content: item, meta}});
+  }
+
+  async get({id, token}) {
+    const {controllerKey, instance, capability} = this;
+    let results;
+    if(id) {
+      results = await findDocuments({controllerKey, instance, id, capability});
+    } else if(token) {
+      results = await findDocuments({
+        controllerKey,
+        instance,
+        equals: {'content.type': this.type, 'meta.token.id': token},
+        capability
+      });
+    } else {
+      throw new TypeError('"id" or "token" must be given.');
+    }
+    if(results.length > 0) {
+      return results[0];
+    }
+    return null;
+  }
+
+  async getAll() {
+    const {controllerKey, instance, capability} = this;
+    const results = await findDocuments(
+      {controllerKey, instance, type: this.type, capability});
+    return results;
+  }
+
+  async update({item, meta, sequence}) {
+    if(!item || (item.type !== this.type)) {
+      throw new TypeError(`"item.type" (${item.type}) must be "${this.type}".`);
+    }
+    const {controllerKey, instance, capability} = this;
+    const existing = await this.get({id: item.id});
+    const edvDoc = await getEdvDocument(
+      {controllerKey, instance, id: existing.id, capability});
+    const doc = await edvDoc.read();
+    if(sequence !== undefined && sequence !== doc.sequence) {
+      const err = new BedrockError(
+        'Could not update item; a conflict occurred.',
+        'InvalidStateError');
+      throw err;
+    }
+    const updatedDoc = {
+      ...doc
+    };
+    if(meta) {
+      updatedDoc.meta = meta;
+    }
+    await edvDoc.write({
+      doc: updatedDoc
+    });
+  }
+
+  async remove({id}) {
+    const {controllerKey, instance, capability} = this;
+    const existing = await this.get({id});
+    if(!existing) {
+      return false;
+    }
+    const edvDoc = await getEdvDocument(
+      {controllerKey, instance, id: existing.id, capability});
+    return edvDoc.delete();
+  }
+};
+
+async function getEdvDocument({controllerKey, instance, id, capability}) {
+  const client = await getEdvClient({controllerKey, instance});
+  const {keyAgreementKey, hmac} = client;
+  const recipients = [{
+    header: {kid: keyAgreementKey.id, alg: 'ECDH-ES+A256KW'}
+  }];
+  const invocationSigner = await controllerKey.getAsymmetricKey(
+    instance.keys.zcapKey);
+  return new EdvDocument({
+    id, recipients, keyResolver, keyAgreementKey, hmac, capability,
+    invocationSigner, client
+  });
+}
+
+async function getEdvClient({controllerKey, instance}) {
+  const [keyAgreementKey, hmac] = await Promise.all([
+    controllerKey.getKeyAgreementKey(instance.keys.kak),
+    controllerKey.getHmac(instance.keys.hmac)
+  ]);
+  // TODO: base this off of bedrock.config
```

Comment:
this is address by using: https://github.com/digitalbazaar/bedrock-https-agent

Reply:
@mattcollier -- if you know how to fix, can you do so or file an issue for it?

Reply:
self-assigned issue